src/components/Album.js

```diff
@@ -30,82 +35,332 @@ import AddToWishlistButton from './layout/AddToWishlistButton/AddToWishlistButto
 
 function Album({ album }) {
   const [openDialog, setOpenDialog] = useState(false);
+  const [reviewList, setReviewList] = useState([]);
+  const [formData, setFormData] = useState({
+    title: '',
+    comment: '',
+    rating: '',
+  });
+  const [isEdit, setIsEdit] = useState('');
+  const [editObj, setEditObj] = useState('');
+  const [loginID, setLoginID] = useState('');
 
-  const handleOpenDialog = () => {
-    setOpenDialog(true);
+  const { status } = useAuth();
+  const isLoggedIn = status === AuthStatus.LoggedIn;
+
+  console.log('isLoggedIn=====', isLoggedIn)
+
+  const handleOpenDialog = async () => {
+    try {
+      setOpenDialog(true);
+      const response = await axiosInstance.get(`/reviews/album/${album._id}`);
+      setReviewList(response.data.allProductReviews);
+    } catch (error) {
+      console.error('Error fetching review:', error);
+    }
   };
 
   const handleCloseDialog = () => {
     setOpenDialog(false);
+    setIsEdit('');
+    setEditObj('');
+    setFormData({
+      title: '',
+      comment: '',
+      rating: '',
+    });
+  };
+
+  const show = () => {
+    const data = reviewList.filter((val) => val.user === loginID);
+    return data.length === 0 && isLoggedIn;
+  };
```

Comment:
This is fine as-is but here is a suggestion for making a it a one-line variable rather than a function:

```suggestion
  const show = isLoggedIn && !reviewList.some((val) => val.user === loginID);
```

  
[Array.some ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) will return true if the boolean operation inside returns true for at least one element in the list. We want to know if _none_ of the elements in the list belong to the user so we negate with `!`